File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cs
using System;

public class Solution {
    public int solution(string myString, string pat) 
    {
        myString = myString.Replace("A", "b")
                           .Replace("B", "a");
        
        return myString.Contains(pat.ToLower()) ? 1 : 0;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    
    string newPat;
    for (const char& c : pat)
    {
        if (c == 'A')
            newPat.push_back('B');
        else if (c == 'B')
            newPat.push_back('A');
        else
            newPat.push_back(c);
    }
    
    size_t pos = myString.find(newPat);
    return pos != std::string::npos ? 1 : 0;
}